File: 0063 - Unique Paths II/solution.c
int uniquePathsWithObstacles(int** grid, int obstacleGridSize, int* obstacleGridColSize){
    int m = obstacleGridSize, n = *obstacleGridColSize;
    int b[105];
    b[0] = grid[0][0]?0:1;
    for(int j = 1; j < n; j++) b[j] = grid[0][j]?0:b[j-1];
    for(int i = 1; i < m; i++){
        b[0] = grid[i][0]?0:b[0];
        for(int j = 1; j < n; j++){
            b[j] = grid[i][j]?0:b[j]+b[j-1];
        }
    }
    return b[n-1];
}

File: 0063 - Unique Paths II/solution.py
class Solution:
    def uniquePathsWithObstacles(self, A: List[List[int]]) -> int:
        m, n = len(A), len(A[0])
        dp = [0, 1]+[0]*(n-1);
        for i in range(1, m+1):
            for j in range(1, n+1):
                dp[j] = 0 if A[i-1][j-1] else dp[j]+dp[j-1]
        return dp[-1]